src/main/groovy/com/twcable/grabbit/jcr/ProtoNodeDecorator.groovy

```diff
@@ -91,14 +91,22 @@ class ProtoNodeDecorator {
      * @param property
      * @param node
      */
-    private static void addMixins(ProtoPropertyDecorator property, JCRNode node) {
+    private static void addMixins(ProtoPropertyDecorator property, JCRNode node, Session session) {
         property.values.valueList.each { ProtoValue value ->
-            if (node.canAddMixin(value.stringValue)) {
-                node.addMixin(value.stringValue)
-                log.debug "Added mixin ${value.stringValue} for : ${node.name}."
+            try{
+                if(!node.isCheckedOut()){
+                    session.getWorkspace().getVersionManager().checkout(node.path)
+                }
```

Comment:
Actually, thinking about this more, I am wondering if we should be preventing the case for "node is already checked in" in the first place. 

That is, I think there might be an opportunity to check how the server is serializing the nodes and may be there is an issue with how we are serializing the nodes for some case which is leading to the "node is checked in" error on the client.